### ISSUE TITLE
Ktor + Bolt + Events throwing "a request for an unknown workspace detected" w/ MultiTeamAuthorization middleware

### ISSUE THREAD

**Issue by BradleyIW** [slackapi/java-slack-sdk#845]
I've gone through the OAuth 2.0 flow as directed on the website, i'm using Ktor to achieve this. The installation to the workspace seems fine via my /slack/install endpoint but when I go to make an events call after the fact the events API request returns an error message (once parsed using the `toBoltRequest` and `respond` suspend functions): 

```
2021-09-10T06:43:12.700447+00:00 app[web.1]: 2021-09-10 06:43:12.700 [eventLoopGroupProxy-4-7] INFO  c.s.a.b.m.b.MultiTeamsAuthorization - auth.test result: null, io error: null, api error: null
2021-09-10T06:43:12.700474+00:00 app[web.1]: 2021-09-10 06:43:12.700 [eventLoopGroupProxy-4-7] DEBUG Application - This is the response body for /slack/events {"error":"a request for an unknown workspace detected"}
```

I have the default File installation service + state service attached (for now) to my App object. Inside the MultiTeamsAuthorization class it seems to be hitting line https://github.com/slackapi/java-slack-sdk/blob/9d34fe649e899cb0f868b5dc05370eacb2b2466b/bolt/src/main/java/com/slack/api/bolt/middleware/builtin/MultiTeamsAuthorization.java#L225

I must be missing something small as the request to the events api itself is fine, it's when we parse it using the helper classes `toBoltRequest` and `respond` it'll respond with a 401.  

Any help would be appreciated - documentation regarding Ktor seems to be slim. 

### Reproducible in:

```bash
mvn dependency:tree | grep com.slack.api
gradle dependencies | grep com.slack.api
java -version
sw_vers && uname -v # or `ver`
```

#### The Slack SDK version

+--- com.slack.api:bolt-ktor:1.12.1
|    +--- com.slack.api:slack-api-client:1.12.1
|    |    +--- com.slack.api:slack-api-model:1.12.1
|    +--- com.slack.api:bolt:1.12.1
|    |    +--- com.slack.api:slack-api-model:1.12.1 (*)
|    |    +--- com.slack.api:slack-api-client:1.12.1 (*)
|    |    \--- com.slack.api:slack-app-backend:1.12.1
|    |         +--- com.slack.api:slack-api-model:1.12.1 (*)
|    |         +--- com.slack.api:slack-api-client:1.12.1 (*)

#### Java Runtime version

openjdk 14.0.2 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 14.0.2+12)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 14.0.2+12, mixed mode, sharing)

#### OS info

ProductName: macOS
ProductVersion: 11.5.2
BuildVersion:   20G95
Darwin Kernel Version 20.6.0: Wed Jun 23 00:26:31 PDT 2021; root:xnu-7195.141.2~5/RELEASE_X86_64

#### Steps to reproduce:

(Share the commands to run, source code, and project settings (e.g., pom.xml/build.gradle))

1.
2.
3.

#### Expected result:

Once request has been parsed, it should run a slash command 

#### Actual result:

```
2021-09-10T06:43:12.700447+00:00 app[web.1]: 2021-09-10 06:43:12.700 [eventLoopGroupProxy-4-7] INFO  c.s.a.b.m.b.MultiTeamsAuthorization - auth.test result: null, io error: null, api error: null
2021-09-10T06:43:12.700474+00:00 app[web.1]: 2021-09-10 06:43:12.700 [eventLoopGroupProxy-4-7] DEBUG Application - This is the response body for /slack/events {"error":"a request for an unknown workspace detected"}
```

### Requirements

Please make sure if this topic is specific to this SDK. For general questions/issues about Slack API platform or its server-side, could you submit questions at https://my.slack.com/help/requests/new instead. :bow:

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to the those rules.

**Comment by seratch** [comment]
Hi @BradleyIW, thanks for asking the question! 

Just in case, I've verified the behavior with [this example](https://github.com/slackapi/java-slack-sdk/blob/main/bolt-ktor/src/test/kotlin/examples/oauth/app.kt) but it works for me without any issues. The situation is that your app does not have saved installation data for the incoming requests (=InstallationService was not able to find corresponding installation data for the workspace / enterprise grid org).

I would recommend trying [this simple example](https://github.com/slackapi/java-slack-sdk/blob/main/bolt-ktor/src/test/kotlin/examples/oauth/app.kt) first. Also, the default file based installation service saves data under `$HOME/.slack-app/`. Checking the directory may be helpful to figure out the reason of your situation.

I hope this helps!

**Comment by BradleyIW** [comment]
Hi @seratch, 

Thanks for the speedy response, the information around the `InstallationService` solved this for me. I found that the `FileInstallationService` wasn't suitable for the service I was hosting the bot on, so it wasn't storing the installation data (as you suspected). I created a custom `InstallationService` using a relational database and now it works seamlessly. 

I'll close this issue, thanks again.
